src/test/java/core/basesyntax/service/RegistrationServiceImplTest.java

```diff
@@ -0,0 +1,146 @@
+package core.basesyntax.service;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNull;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+import core.basesyntax.dao.StorageDao;
+import core.basesyntax.dao.StorageDaoImpl;
+import core.basesyntax.db.Storage;
+import core.basesyntax.exceptions.RegistrationException;
+import core.basesyntax.model.User;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+class RegistrationServiceImplTest {
+    private static final String VALID_LOGIN = "validLogin";
+    private static final String VALID_PASSWORD = "validPassword";
+    private static final String EXISTING_USER_LOGIN = "existingUserLogin";
+    private static final String SHORT_LOGIN = "log";
+    private static final String SHORT_PASSWORD = "pass";
+    private static final String EMPTY_STRING = "";
+    private static final int VALID_AGE = 24;
+    private static final int INVALID_AGE = 15;
+
+    private RegistrationService registrationValidator;
+    private StorageDao storageDao;
+
+    @BeforeEach
+    void setUp() {
+        Storage.people.clear();
+        registrationValidator = new RegistrationServiceImpl();
+        storageDao = new StorageDaoImpl();
+    }
+
+    @Test
+    public void register_existingUser_notOk() {
+        User existingUser = createUser(EXISTING_USER_LOGIN, VALID_PASSWORD, VALID_AGE);
+        storageDao.add(existingUser);
+        assertThrows(RegistrationException.class, ()
+                -> registrationValidator.register(existingUser));
+    }
```

Comment:
The test `register_existingUser_notOk` does not actually test the scenario where a user with an existing login attempts to register again. It only tests that an exception is thrown when trying to register the same user object twice. You should add a test case where you attempt to register a new user with the same login as an existing user to properly test this case.